app/controllers/admin/ex_links_controller.rb

```diff
@@ -0,0 +1,146 @@
+# encoding:UTF-8
+class Admin::ExLinksController < ApplicationController
+  load_permissions_and_authorize_resource
+  before_action :authorize
+  before_action :set_ex_link, only: [:show, :edit, :update, :destroy]
+
+  # GET /ex_links
+  def index
+    if params[:tag]
+      @ex_links = get_exlinks_if_all_tags_present(params[:tag])
+    else
+      @ex_links = ExLink.all
+    end
+    @tags = get_tags
+  end
+
+  # GET /ex_links/1
+  def show
+  end
+
+  # GET /ex_links/new
+  def new
+    @ex_link = ExLink.new
+  end
+
+  # GET /ex_links/1/edit
+  def edit
+  end
+
+  # POST /ex_links
+  def create
+    @ex_link = ExLink.new(ex_link_params)
+    if @ex_link.save
+      redirect_to admin_ex_link_path(@ex_link), notice: alert_create(ExLink)
+    else
+      render :new
```

Comment:
When `create` and `update` fail it should return a `422: unprocessable entity`.
It can be done with: `render :new, status: 422`